.github/workflows/test.yml

```diff
@@ -0,0 +1,85 @@
+name: Test
+on:
+  push:
+    branches:
+    - main
+    - feature/datasets
+    - tz/pypi
+  pull_request:
+    branches:
+    - main
+    - feature/datasets
+jobs:
+  test:
+    name: ${{ matrix.lang }} tests on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        lang: [Python]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
+    - name: Install Poetry
+      run: |
+        pip install --upgrade pip
+        pip install poetry coveragepy-lcov
+
+    - name: Cache Poetry virtualenv
+      uses: actions/cache@v1
+      id: cache
+      with:
+        path: ~/.virtualenvs
+        key: poetry-${{ hashFiles('**/poetry.lock') }}
+        restore-keys: |
+          poetry-${{ hashFiles('**/poetry.lock') }}
+
+    - name: Set Poetry config
+      run: |
+        poetry config virtualenvs.in-project false
+        poetry config virtualenvs.path ~/.virtualenvs
+
+    - name: Install Dependencies
+      run: |
+        poetry install -E dask -E spark
+        poetry run pip install --upgrade git+https://github.com/zillow/metaflow.git@feature/kfp
```

Comment:
How come the tests rely on the `feature/kfp` branch and not the `-E metaflow`? Guessing that's because we don't publish a built `zillow-metaflow` library artifact anywhere accessible within reach of this Open Source repo?

Reply:
Exactly and unfortunately!